refined/shared/src/main/scala/monocle/refined/strings.scala

```diff
@@ -8,16 +8,17 @@ object strings extends StringsInstances
 
 trait StringsInstances {
 
-  def startsWith(prefix: String): Prism[String, StartsWithString[prefix.type]] = {
-    Prism.partial[String, Refined[String, StartsWith[prefix.type]]] {
-      case string if string.startsWith(prefix) => Refined.unsafeApply(string)
-    }{_.value}
-  }
+  def startsWith(prefix: String)(implicit v: Validate[String, StartsWith[prefix.type]]): Prism[String, StartsWithString[prefix.type]] =
+    refinedPrism[String, StartsWith[prefix.type]](prefix)
 
-  def endsWith(suffix: String)(implicit v: Validate[String, EndsWith[suffix.type]]): Prism[String, EndsWithString[suffix.type]] = {
-    Prism.partial[String, Refined[String, EndsWith[suffix.type]]] {
-      case string if v.isValid(string) => Refined.unsafeApply(string)
-    }{_.value}
-  }
+  def endsWith(suffix: String)(implicit v: Validate[String, EndsWith[suffix.type]]): Prism[String, EndsWithString[suffix.type]] =
+    refinedPrism[String, EndsWith[suffix.type]](suffix)
 
+  private def refinedPrism[T, P](t: T)(implicit v: Validate[T, P]): Prism[T, T Refined P] = {
+    Prism.partial[T, Refined[T, P]] {
+      case tt if v.isValid(t) => Refined.unsafeApply[T, P](tt)
```

Comment:
We need to check if `tt` is valid since this is the value we are wrapping with `Refined`. The `t` parameter here will be `prefix` and `suffix` from above and therefore `v.isValid(t)` will always be true. The `t` parameter is not needed to define `refinedPrism`. If you remove that parameter, you can also use `refinedPrism` to implement `chars.lowerCase` and `chars.upperCase` with it.